### PR TITLE
Gauge: Better handling of gauge repeat title

### LINKED ISSUE #15244: Gauge multi series mode
If the gauge panel gets multiple series it should repeat itself (the gauge) in a horizontal or vertical direction. 

New options
* Multi series mode: Repeat or combine

Repeat mode
* Direction: auto, vertical or horizontal. 

In repeat mode, each gauge will need a title (series name) above the gauge. 

The thinking here is that this multi-series/repeat behavior & options is going to be the same for singlestat, gauge, bar gauge, traffic light (or some other single stat viz)

### PR DESCRIPTION
**Pull request by torkelo**
When Gauge is repeating (due to query returning many series) it shows a title (name for each series or field).

This PR aims to improve positioning and font size of that title.
Closes #15244

### CODE REVIEW

File: packages/grafana-ui/src/components/Gauge/Gauge.tsx [hunk 1]
```diff
@@ -64,13 +63,14 @@ export class Gauge extends PureComponent<Props> {
   draw() {
     const { maxValue, minValue, showThresholdLabels, showThresholdMarkers, width, height, theme, value } = this.props;
 
-    const dimension = Math.min(width, height * 1.3);
-    const backgroundColor = theme.type === GrafanaThemeType.Light ? 'rgb(230,230,230)' : theme.colors.dark3;
+    const autoProps = calculateGaugeAutoProps(width, height, value.title);
+    const dimension = Math.min(width, autoProps.gaugeHeight);
 
+    const backgroundColor = theme.type === GrafanaThemeType.Light ? 'rgb(230,230,230)' : theme.colors.dark3;
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Gauge/Gauge.tsx]: we can change this to `selectThemeVariant`

File: packages/grafana-ui/src/components/Gauge/Gauge.tsx [hunk 2]
```diff
@@ -64,13 +63,14 @@ export class Gauge extends PureComponent<Props> {
...
     const gaugeWidthReduceRatio = showThresholdLabels ? 1.5 : 1;
-    const gaugeWidth = Math.min(dimension / 6, 60) / gaugeWidthReduceRatio;
+    const gaugeWidth = Math.min(dimension / 6, 40) / gaugeWidthReduceRatio;
     const thresholdMarkersWidth = gaugeWidth / 5;
-    const fontSize = Math.min(dimension / 5, 100) * (value.text !== null ? this.getFontScale(value.text.length) : 1);
+    const fontSize = Math.min(dimension / 5.5, 100) * (value.text !== null ? this.getFontScale(value.text.length) : 1);
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Gauge/Gauge.tsx]: 5 -> 5.5 it there any particular math behind that change or is it "manual test" decision?:)

File: packages/grafana-ui/src/components/Gauge/Gauge.tsx [hunk 3]
```diff
@@ -148,3 +161,23 @@ export class Gauge extends PureComponent<Props> {
...
 }
+
+interface GaugeAutoProps {
+  titleFontSize: number;
+  gaugeHeight: number;
+  showLabel: boolean;
+}
+
+function calculateGaugeAutoProps(width: number, height: number, title: string | undefined): GaugeAutoProps {
+  const showLabel = title !== null && title !== undefined;
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Gauge/Gauge.tsx]: detail - typing says `title` can't be null so I guess this check is unnecessary

File: packages/grafana-ui/src/components/Gauge/Gauge.tsx [hunk 4]
```diff
@@ -148,3 +161,23 @@ export class Gauge extends PureComponent<Props> {
...
   }
 }
+
+interface GaugeAutoProps {
+  titleFontSize: number;
+  gaugeHeight: number;
+  showLabel: boolean;
+}
+
+function calculateGaugeAutoProps(width: number, height: number, title: string | undefined): GaugeAutoProps {
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Gauge/Gauge.tsx]: `title?: string`

**Review by dprokop — COMMENTED** [review]
Minor stuff, apart from that 👍

File: packages/grafana-ui/src/components/Gauge/Gauge.tsx (hunk 4 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/Gauge/Gauge.tsx]: It’s not an optional argument, but I guess for typescript that is the same

File: packages/grafana-ui/src/components/Gauge/Gauge.tsx (hunk 2 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/Gauge/Gauge.tsx]: Yea, value font size was too big in some cases

File: packages/grafana-ui/src/components/Gauge/Gauge.tsx (hunk 1 above)
Comment (dprokop) [reply packages/grafana-ui/src/components/Gauge/Gauge.tsx]: changed

**Pull request merged**